lib/collection/src/common/snapshots_manager.rs

```diff
@@ -32,11 +66,35 @@ pub enum SnapshotStorageManager {
 }
 
 impl SnapshotStorageManager {
-    pub fn new(s3_config: Option<S3Config>) -> Self {
-        if let Some(s3_config) = s3_config {
-            SnapshotStorageManager::S3(SnapshotStorageS3 { s3_config })
-        } else {
-            SnapshotStorageManager::LocalFS(SnapshotStorageLocalFS)
+    pub async fn new(snapshots_config: SnapShotsConfig) -> Self {
+        match snapshots_config.clone().snapshots_storage {
+            SnapshotsStorageConfig::Local => {
+                SnapshotStorageManager::LocalFS(SnapshotStorageLocalFS)
+            }
+            SnapshotsStorageConfig::S3 => {
+                if let Some(s3_config) = &snapshots_config.s3_config {
+                    // Set AWS credentials to environment variables from the config
+                    if let Some(access_key) = &s3_config.access_key {
+                        env::set_var("AWS_ACCESS_KEY_ID", access_key);
+                    }
+                    if let Some(secret_key) = &s3_config.secret_key {
+                        env::set_var("AWS_SECRET_ACCESS_KEY", secret_key);
+                    }
+                    if let Some(region) = &s3_config.region {
+                        env::set_var("AWS_REGION", region);
+                    }
+                    if let Some(endpoint_url) = &s3_config.endpoint_url {
+                        env::set_var("AWS_ENDPOINT_URL", endpoint_url);
+                    }
```

Comment:
This looks a bit artificial. Why can't we just directly pass parameters to the constructor?